six/modules/c++/samples/check_valid_six.cpp

```diff
@@ -148,16 +148,26 @@ int main(int argc, char** argv)
             log->info(Ctxt("Reading " + inputPathname));
             try
             {
-                if (nitf::Reader::getNITFVersion(inputPathname) ==
-                        NITF_VER_UNKNOWN)
+                std::auto_ptr<six::Data> data =
```

Comment:
This doesn't work right if I send in a NITF - `parseDataFromFile()` is counting on getting in an XML text file